contrib/core-contract-tests/tests/pox-4/pox_DelegateStackExtendCommand.ts

```diff
@@ -0,0 +1,160 @@
+import {
+  logCommand,
+  PoxCommand,
+  Real,
+  Stub,
+  Wallet,
+} from "./pox_CommandModel.ts";
+import { poxAddressToTuple } from "@stacks/stacking";
+import { assert, expect } from "vitest";
+import {
+  Cl,
+  ClarityType,
+  isClarityType,
+} from "@stacks/transactions";
+import {
+  FIRST_BURNCHAIN_BLOCK_HEIGHT,
+  REWARD_CYCLE_LENGTH,
+} from "./pox_Commands.ts";
+
+/**
+ * The `DelegateStackExtendCommand` allows a pool operator to
+ * extend an active stacking lock, issuing a "partial commitment" 
+ * for the extended-to cycles.
+ * 
+ * This method extends stacker's current lockup for an additional 
+ * extend-count and partially commits those new cycles to `pox-addr`.
+ *
+ * Constraints for running this command include:
+ * - Stacker must have locked uSTX.
+ * - The Operator has to currently be delegated by the Stacker.
+ * - The new lock period must be less than or equal to 12.
+ */
+export class DelegateStackExtendCommand implements PoxCommand {
+  readonly operator: Wallet;
+  readonly stacker: Wallet;
+  readonly extendCount: number;
+  readonly currentCycle: number;
+
+  /**
+   * Constructs a `DelegateStackExtendCommand` to extend the unlock
+   * height as a Pool Operator on behalf of a Stacker.
+   *
+   * @param operator - Represents the Pool Operator's wallet.
+   * @param stacker - Represents the STacker's wallet.
+   * @param extendCount - Represents the cycles to be expended.
+   * @param currentCycle - Represents the current PoX reward cycle.
+   */
+  constructor(
+    operator: Wallet,
+    stacker: Wallet,
+    extendCount: number,
+    currentCycle: number,
+  ) {
+    this.operator = operator;
+    this.stacker = stacker;
+    this.extendCount = extendCount;
+    this.currentCycle = currentCycle;
+  }
+
+  check(model: Readonly<Stub>): boolean {
+    // Constraints for running this command include:
+    // - Stacker must have locked uSTX.
+    // - The Stacker's uSTX must have been locked by the Operator.
+    // - The Operator has to currently be delegated by the Stacker.
+    // - The new lock period must be less than or equal to 12.
+
+    const operatorWallet = model.wallets.get(this.operator.stxAddress)!;
+    const stackerWallet = model.wallets.get(this.stacker.stxAddress)!;
+
+    const firstRewardCycle =
+      this.currentCycle > this.stacker.firstLockedRewardCycle
+        ? this.currentCycle
+        : this.stacker.firstLockedRewardCycle;
+    const firstExtendCycle = Math.floor(
+      (this.stacker.unlockHeight - FIRST_BURNCHAIN_BLOCK_HEIGHT) /
+        REWARD_CYCLE_LENGTH,
+    );
+    const lastExtendCycle = firstExtendCycle + this.extendCount - 1;
```

Comment:
Why do you need this value? You are substracting one and adding one afterwards.

Reply:
@friedger I am doing this to simulate how the lock period is [calculated](https://github.com/stacks-network/stacks-core/blob/next/stackslib/src/chainstate/stacks/boot/pox-4.clar#L1328-L1330) inside PoX-4 contract